resources/todomvc/big-dom-generator/src/components/tree-area.jsx

```diff
@@ -1,56 +1,68 @@
-import { LCG } from "random-seedable";
-import { useRef } from "react";
-import { DEFAULT_SEED_FOR_RANDOM_NUMBER_GENERATOR, MAX_GENERATED_DOM_DEPTH, MAX_NUMBER_OF_CHILDREN, PROBABILITY_OF_HAVING_CHILDREN, TARGET_SIZE, MAX_VISIBLE_TREE_VIEW_ITEM_DEPTH } from "../../params";
+import { MAX_VISIBLE_TREE_VIEW_ITEM_DEPTH } from "../../params";
+import { generateTreeHead } from "./../tree-generator";
 
 import ChevronRight from "./../assets/Smock_ChevronRight_18_N.svg";
 import TaskListIcon from "./../assets/Smock_TaskList_18_N.svg";
 
-const FolderWrapper = (props) => {
-    const { nodeCount, random, maxDepth, maxNumChildren, childProbability, currentDepth } = props;
-    // prettier-ignore
-    if (currentDepth >= maxDepth)
-        return null;
-    // Choose a random number of children.
-    const numChildren = random.randRange(1, maxNumChildren);
-    const children = [];
-    for (let i = 0; i < numChildren && nodeCount.current < TARGET_SIZE; i++)
-        children.push(<TreeItem key={i} nodeCount={nodeCount} random={random} numChildren={numChildren} maxNumChildren={maxNumChildren} maxDepth={maxDepth} childProbability={childProbability} currentDepth={currentDepth + 1} />);
-
-    nodeCount.current = nodeCount.current + 1;
-    return <ul className="spectrum-TreeView spectrum-TreeView--sizeS">{children}</ul>;
-};
-
 const TreeItem = (props) => {
-    const { nodeCount, random, maxDepth, maxNumChildren, childProbability, currentDepth } = props;
-    nodeCount.current = nodeCount.current + 4;
-    // Choose whether to have children.
-    const children = random.coin(childProbability) ? <FolderWrapper nodeCount={nodeCount} random={random} maxNumChildren={maxNumChildren} maxDepth={maxDepth} childProbability={childProbability} currentDepth={currentDepth + 1} /> : null;
-    const treeViewItemIsOpen = children && currentDepth < MAX_VISIBLE_TREE_VIEW_ITEM_DEPTH ? "is-open" : "";
+    const { treeNode, currentDepth } = props;
+
+    const isExpandableItem = treeNode.type === "expandableItem";
+    const children = isExpandableItem ? treeNode.children.map((child) => <TreeItem treeNode={child} currentDepth={currentDepth + 1} />) : null;
+    const treeViewItemIsOpen = isExpandableItem && currentDepth < MAX_VISIBLE_TREE_VIEW_ITEM_DEPTH ? "is-open" : "";
     return (
         <li className={`spectrum-TreeView-item ${treeViewItemIsOpen}`}>
```

Comment:
can you please add a class for the type of item? eg `nodetype-${treeNode.type}`. This makes it easier to look at and debug the rendered DOM later.

Reply:
Done.